### PR TITLE
More provenance specs + logic improvements

### PR DESCRIPTION
**Pull request by ben-bay**
Inside `merlin_info` we now have `<name>.expanded.yaml`, `<name>.orig.yaml`, and `<name>.partial.yaml`. The latter has only expanded its user variables section.

I would appreciate a little extra testing for this one.

### CODE REVIEW

File: merlin/study/study.py [hunk 1]
```diff
@@ -116,33 +117,44 @@ def __init__(
         self.dag = None
         self.load_dag()
 
+    @cached_property
+    def provenance_original_spec(self):
+        return os.path.join(self.info, "orig.yaml")
```
Comment (koning) [inline merlin/study/study.py:122]: Maybe prepend orig_ to the filename. orig_filename

File: merlin/study/study.py [hunk 2]
```diff
@@ -353,8 +363,9 @@ def expanded_spec(self):
         )
 
         # expand provenance spec filename
-        if "$(" in self.spec.name:
-            expanded_name = result.description["name"].replace(" ", "_") + ".yaml"
+        if "$(" in self.original_spec.name:
+            # expanded_name = result.description["name"].replace(" ", "_") + ".yaml"
+            expanded_name = "expanded.yaml"
```
Comment (koning) [inline merlin/study/study.py:368]: We should preserve the original filename.

File: merlin/study/study.py [hunk 3]
```diff
@@ -161,16 +173,16 @@ def write_expanded_spec(self, dest):
...
-        full_spec = dump_with_overrides(self.spec, self.override_vars)
+        full_spec = dump_with_overrides(self.original_spec, self.override_vars)
 
         with open(dest, "w") as dumped_file:
             dumped_file.write(full_spec)
 
-        # update spec so that user_vars update will be accurate
-        self.spec = MerlinSpec.load_specification(dest)
+        # load spec so that user_vars update will be accurate
+        new_spec = MerlinSpec.load_specification(dest)
```
Comment (koning) [inline merlin/study/study.py:182]: Should we save this spec?

File: merlin/study/study.py (hunk 1 above)
Comment (lucpeterson) [reply merlin/study/study.py:122]: yeah i think this is a good idea. which do we prefer?

prepend (orig_name.yaml)
or insert near the end (name_orig.yaml)

Comment (koning) [reply merlin/study/study.py:122]: Whichever is easier.

File: merlin/study/study.py (hunk 3 above)
Comment (ben-bay) [reply merlin/study/study.py:180]: No, we're just using it two lines down.

**Comment by ben-bay** [comment]
@koning @lucpeterson okay I have fixed the bugs that were here previously, and all tests are passing. The `Study` class is big and is starting to show some bad smells, so at some point I'd like to refactor the whole thing. But for now this appears to deliver the functionality we need.

**Pull request merged**